internal/server/boltdbstate/pipeline.go

```diff
@@ -213,6 +213,11 @@ func (s *State) PipelineList(pRef *pb.Ref_Project) ([]*pb.Pipeline, error) {
 			if err != nil {
 				return err
 			}
+			if pRef != nil {
+				if val.Owner.(*pb.Pipeline_Project).Project.Project != pRef.Project {
+					continue
+				}
+			}
```

Comment:
This makes sense, but should we instead make `project` an index on pipelines (it might be already) and fix the query in lower case `pipelineList` so it'll only return a list of pipelines scoped to the `Ref_Project`? 🤔

Reply:
This pattern actually matches the pattern in JobList, where we just grab all the jobs and then filter out the ones we don't need.

Reply:
So we definitely could add a `listPipelineByProject` method, but I wouldn't want to add filtering into the db method since I think those should be kept as clean and single-purpose as possible. Also keeping things consistent in the codebase is good :)

Reply:
Oh yeah, I was mostly meaning make `PipelineList` _only_ ever scoped to a project since it's already accepting a project ref in its parameters. We don't list pipelines across all projects right now so I was assuming that's how it should work! I wouldn't add another method though at this time. The index query would be faster than a loop too. We do already index on Project it looks like too so should be an easy add! 

https://github.com/hashicorp/waypoint/blob/7a1f832cf8a6cc8e3eac869ce9a135dbd4bef1cc/internal/server/boltdbstate/pipeline.go#L302-L310

Reply:
Oh I see what you mean, and we could add a `PipelineListAll` in the future if we ever want cross-project pipeline listing.

Reply:
Yep, exactly!